server.js

```diff
@@ -3838,12 +3839,23 @@ cache(function(data, match, sendBadge, request) {
         return;
       }
       var body = JSON.parse(buffer);
-      if (body.license != null) {
-        badgeData.text[1] = body.license.name;
-        badgeData.colorscheme = 'blue';
+      const license = body.license;
+      if (license != null) {
+        if (license.spdx_id) {
+          badgeData.text[1] = license.spdx_id;
+        } else {
+          badgeData.text[1] = 'unknown';
+        }
+        const color = licenseToColor(license.spdx_id);
+        if (sixHex(color)) {
+          badgeData.colorB = '#' + color;
+        } else {
+          badgeData.colorscheme = color;
+        }
```

Comment:
What do you think about adding a helper function in `badge-data`, so these few lines can be written like this?
```js
setBadgeColor(badgeData, licenseToColor(license.spdx_id));
```
It's reads bit clearer that way.

Reply:
done